backend/JiperBackend/Controllers/UserController.cs

```diff
@@ -142,5 +143,83 @@ public IActionResult GetOrder(int userId, int orderId)
             }
             return Ok(order);
         }
+
+        [HttpPost("updateinfo")]
+        public IActionResult UpdateInfo([FromBody] JObject data)
+        {
+            User user;
+            try
+            {
+                int userId = data["userId"].ToObject<int>();
+                user = userService.GetUser(userId) ?? throw new ArgumentNullException();
+                user.Email = data["email"].ToObject<string>();
+                user.Password = data["password"].ToObject<string>();
+                user.FirstName = data["firstName"].ToObject<string>();
+                user.LastName = data["lastName"].ToObject<string>();
+                user.PhoneNumber = data["phoneNumber"].ToObject<string>();
+                Address address = data["address"].ToObject<Address>();
+                user.Address.City = address.City;
+                user.Address.Street = address.Street;
+                user.Address.HouseNr = address.HouseNr;
+                user.Address.ZipCode = address.ZipCode;
+                userService.SaveChanges();
+            }
+            catch (NullReferenceException)
+            {
+                return BadRequest();
+            }
+            catch (ArgumentNullException)
+            {
+                return NotFound();
+            }
+            catch (DbUpdateConcurrencyException)
+            {
+                return Conflict("OPTIMISTIC_LOCK_ERROR");
+            }
+            return Ok(user);
+        }
+
+        [HttpPost("resolveconflict")]
+        public IActionResult ResolveConflict([FromBody] JObject data)
```

Comment:
I am not sure if "resolveconflict" is a good name for endpoint.

Reply:
Then what is a good name? This endpoint literally resolves DB conflict. "updateinfosecondattempt"? :/